pex/pex_builder.py

```diff
@@ -119,10 +119,12 @@ def __entry_point_from_filename__(filename):
 if '__file__' in locals() and __file__ is not None and os.path.exists(__file__):
   __entry_point__ = __entry_point_from_filename__(__file__)
 elif '__loader__' in locals():
-  from pkgutil import ImpLoader
   if hasattr(__loader__, 'archive'):
     __entry_point__ = __loader__.archive
-  elif isinstance(__loader__, ImpLoader):
+  elif hasattr(__loader__, 'get_filename'):
```

Comment:
Although it's still true Pex contains vendored code that uses ImpLoader (inside Pip), this adjustment as well as the support for find_spec / [PEP-451](https://peps.python.org/pep-0451) below is enough to close #2137.